database/DataGenerator/createHouseData.js

```diff
@@ -0,0 +1,247 @@
+        const house_id = i;
+        const guests = faker.random.number({ min: 1, max: 8 });
+        const bath = faker.random.number({ min: 1, max: 3 });
+
+        if(!writeStream.write(`${i},${house_id},${guests},${bath}`)) {
+          await new Promise(resolve => writeStream.once('drain', resolve));
+        }
+        if (i !== length) {
+          writeStream.write('\n');
+        }
+      }
+      writeStream.end();
+    })();
+    writeStream.on('finish', () => resolve('private rooms generated'));
+  });
+};
+
+const generateBedSizes = (filepath) => {
+  return new Promise((resolve, reject) => {
+
+    const writeStream = fs.createWriteStream(filepath, { encoding: 'utf8' });
+    writeStream.write('id,size\n');
+
+    (async() => {
+      for (let i = 0; i < bedSize.length; i++) {
+        const id = i + 1;
+        if(!writeStream.write(`${id},${bedSize[i].size}`)) {
+          await new Promise(resolve => writeStream.once('drain', resolve));
+        }
+        if (i !== amenityList.length) {
+          writeStream.write('\n');
+        }
+      }
+      writeStream.end();
+    })();
+    writeStream.on('finish', () => resolve('bed size list generated'));
+  });
+};
+
+const generateBedrooms = (filepath, length) => {
+  return new Promise((resolve, reject) => {
+
+    const writeStream = fs.createWriteStream(filepath, { encoding: 'utf8' });
+    writeStream.write('id,private_room_id\n');
+    let id = 1;
+    (async() => {
+      for (let i = 1; i <= length; i++) {
+        const numOfBedrooms = faker.random.number({min: 1, max:3});
+        for (let j = 0; j < numOfBedrooms; j++) {
+          const private_room_id = i;
+
+          if(!writeStream.write(`${id},${private_room_id}`)) {
+            await new Promise(resolve => writeStream.once('drain', resolve));
+          }
+          if (i !== length || j !== numOfBedrooms) {
+            id++;
+            writeStream.write('\n');
+          }
+        }
+      }
+      writeStream.end();
+    })();
+    writeStream.on('finish', () => resolve({message: 'bedrooms generated', bedrooms: id - 1}));
+  });
+};
+
+const generateBeds = (filepath, length) => {
+  return new Promise((resolve, reject) => {
+
+    const writeStream = fs.createWriteStream(filepath, { encoding: 'utf8' });
+    writeStream.write('bedrooms_id,size\n');
+
+    (async() => {
+      for (let i = 1; i <= length; i++) {
+        const numOfBeds = faker.random.number({min: 1, max:3});
+        for (let j = 0; j < numOfBeds; j++) {
+          const bedrooms_id = i;
+          const size = faker.random.number({min: 1, max:3});
+
+          if(!writeStream.write(`${bedrooms_id},${size}`)) {
+            await new Promise(resolve => writeStream.once('drain', resolve));
+          }
+          if (i !== length || j !== numOfBeds) {
+            writeStream.write('\n');
+          }
+        }
+      }
+      writeStream.end();
+    })();
+    writeStream.on('finish', () => resolve('beds generated'));
+  });
+};
+
```

Comment:
@AndrewTjandra I really like how you did the promise chain here since there are so many tables — I might suggest removing some of the console logs or labeling them so they're clearer in the console.